pkg/cli/zip_upload_table_dumps.go

```diff
@@ -356,6 +381,100 @@ func makeTableIterator(f io.Reader) ([]string, func(func(string) error) error) {
 	}
 }
 
+// makeQuotedTSVIterator returns the headers slice and an iterator that properly handles
+// TSV files with quoted fields containing newlines. Unlike makeTableIterator, this
+// function uses readTSVRecord to correctly parse complex fields such as SQL statements
+// that span multiple lines. Returns headers and an iterator function that yields
+// parsed field slices for each record.
+func makeQuotedTSVIterator(f io.Reader) ([]string, func(func([]string) error) error) {
```

Comment:
Can this function be used in all cases? If yes, we should do that and remove the older parser

If not,

1. The signature of the returned `iter` function should be the same in both parsers. You can change either one
2. The comment on this function should clearly state when to use this function. Maybe with an example. It's not fully clear right now.